src/python/pants/backend/adhoc/run_system_binary.py

```diff
@@ -106,6 +110,15 @@ async def _find_binary(
 
     for test, binary in zip(tests, binaries.paths):
         if test.exit_code != 0:
+            if log_fingerprinting_errors:
+                logger.warning(
+                    f"Error occurred while fingerprinting candidate binary `{binary.path}` "
+                    f"for binary `{binary_name}` (exit code {test.exit_code}):\n\n"
```

Comment:
What do you think about referencing the field that controls this to make it super-easy for users to work out what to do? E.g.
```suggestion
                    f"for binary `{binary_name}` (exit code {test.exit_code}) (use the `{SystemBinaryLogFingerprintingErrorsField.alias}` field to control this warning):\n\n"
```